stellarphot/utils/version_migrator.py

```diff
@@ -0,0 +1,174 @@
+            )
+
+        if str(self.to_version.major) not in self.known_versions:
+            raise ValueError(
+                f"Unknown version: {self.to_version=}. "
+                f"Valid versions are: {self.known_versions}"
+            )
+
+        if self.from_version >= self.to_version:
+            raise ValueError(
+                "Can only migrate from a lower version to a higher version"
+            )
+
+        self.camera = camera
+        self.observatory = observatory
+        self.passband_map = passband_map
+
+    @property
+    def from_version(self):
+        """
+        The stellarphot major version to migrate from.
+        """
+        return self._from_version
+
+    @property
+    def to_version(self):
+        """
+        The stellarphot major version to migrate to.
+        """
+        return self._to_version
+
+    def migrate(self, data):
+        """
+        Migrate data from one version to another.
+
+        Parameters
+        ----------
+        data : `astropy.table.Table` or `stellarphot.PhotometryData`
+            The data to migrate.
+        """
+        if self.from_version.major == 1 and self.to_version.major == 2:
+            return self._migrate_v1_v2(data)
+        else:  # pragma: no cover
+            # do not cover because right now no other version combo possible
+            raise ValueError(
+                f"Migration from version {self.from_version} to version "
+                f"{self.to_version} is not supported."
+            )
+
+    def _migrate_v1_v2(self, data):
+        """
+        Migrate data from version 1 to version 2.
+
+        Parameters
+        ----------
+        data : `astropy.table.Table` or `stellarphot.PhotometryData`
+            The data to migrate.
+        """
+        new_data = data.copy()
+
+        unitify = {
+            "fwhm_x": u.pixel,
+            "fwhm_y": u.pixel,
+            "width": u.pixel,
+            "aperture": u.pixel,
+            "annulus_inner": u.pixel,
+            "annulus_outer": u.pixel,
+            "aperture_area": u.pixel,
+            "annulus_area": u.pixel,
+            "noise-aij": u.electron,
+            "annulus_sum": u.adu,
+            "aperture_sum": u.adu,
+            "aperture_net_flux": u.adu,
+            "noise": u.adu,
+            "sky_per_pix_avg": u.adu / u.pixel,
+            "sky_per_pix_med": u.adu / u.pixel,
+            "sky_per_pix_std": u.adu / u.pixel,
+            "RA": u.deg,
+            "Dec": u.deg,
+            "xcenter": u.pixel,
+            "ycenter": u.pixel,
+            "exposure": u.second,
+        }
+        for un in unitify:
+            # Note that this deliberately discards any unit that might have been
+            # present in the original data. This is a stellarphot-specific migration
+            # tool, not something more general, and the choices above are the correct
+            # units for the data.
+            new_data[un] = new_data[un].value * unitify[un]
```

Comment:
Could we potentially have unit conversion issues here?  Or were these units also used in v1 but just not attached to the columns?  I don't remember.

Reply:
The old almost never included units, but it depended on whether the photometry was saved as CSV (no units saved) or FITS (units sometimes present for some things). This is maybe a little risky, but for feder data the image unit was always ADU, times, in seconds, etc.

No one else ever used v1 as far as I know, so this should be ok, I think